boot/zephyr/Kconfig

```
@@ -265,12 +264,26 @@ config BOOT_SERIAL_CDC_ACM
 
 endchoice
 
+choice
+	prompt "Detect type"
+	default BOOT_SERIAL_UART_DETECT
```

```suggestion
	default BOOT_SERIAL_GPIO_DETECT
```

prefer previous detection mechanism as the default one.